packages/table/src/cell/editableCell2.tsx

```diff
@@ -167,9 +174,10 @@ export class EditableCell2 extends React.Component<EditableCell2Props, EditableC
             const textClasses = classNames(Classes.TABLE_EDITABLE_TEXT, {
                 [Classes.TABLE_TRUNCATED_TEXT]: truncated,
                 [Classes.TABLE_NO_WRAP_TEXT]: !wrapText,
+                [Classes.TABLE_CELL_TEXT_PLACEHOLDER]: !hasValue,
             });
 
-            cellContents = <div className={textClasses}>{savedValue}</div>;
+            cellContents = <div className={textClasses}>{hasValue ? savedValue : this.props.placeholder}</div>;
```

Comment:
This is good but I think we need to bring back the change to `cellContents` a few lines above here... `<EditableText placeholder={this.props.placeholder}>` should be set on L168, as suggested in the documentation of the new prop you just added.